moabb/datasets/ssvep_mamem.py

```diff
@@ -2,103 +2,137 @@
+    t_start = np.array(t_start)
+
+    if labels is None:
+        freqs_labels = {6: 1, 7: 2, 8: 3, 9: 4, 11: 5}
+        for f, t in zip(freqs, t_start):
+            eeg[-1, t] = freqs_labels[f]
+    else:
+        for f, t in zip(labels, t_start):
+            eeg[-1, t] = f
+    return eeg
 
 
 class BaseMAMEM(BaseDataset):
     """Base class for MAMEM datasets"""
 
-    def __init__(self, sessions_per_subject, code, doi):
+    def __init__(self, events, sessions_per_subject, code, doi, figshare_id):
         super().__init__(
             subjects=list(range(1, 11)),
-            events={"6.66": 1, "7.50": 2, "8.57": 3, "10.00": 4, "12.00": 5},
+            events=events,
             interval=[1, 4],
             paradigm="ssvep",
             sessions_per_subject=sessions_per_subject,
             code=code,
             doi=doi,
         )
+        self.figshare_id = figshare_id
 
     def _get_single_subject_data(self, subject):
         """return data for a single subject"""
         fnames = self.data_path(subject)
+        filelist = fs_get_file_list(self.figshare_id)
+        fsn = fs_get_file_name(filelist)
         sessions = {}
 
         for fpath in fnames:
+            fnamed = fsn[osp.basename(fpath)]
+            if fnamed[4] == "x":
+                continue
+            session_name = "session_" + fnamed[4]
             if self.code == "SSVEP MAMEM3":
-                fnamed = os.path.basename(fpath)
-                session_name = "session_" + fnamed[4]
                 # Since the data for each session is saved in 2 files,
                 # it is being saved in 2 runs
-                run_number = len(fnamed) - 6
+                run_number = len(fnamed) - 10
                 run_name = "run_" + str(run_number)
             else:
-                session_name = "session_" + fpath[-1]
                 run_name = "run_0"
 
-            record = wfdb.rdrecord(fpath)
-            data = record.p_signal.T
-            annots = wfdb.rdann(fpath, "win")
-            # the number of samples isn't exactly equal in all the trials
-            n_samples = record.sig_len
-            stim_freq = np.array([float(e) for e in self.event_id.keys()])
-            # aux_note are the exact frequencies, matched to nearest class
-            events_label = [
-                np.argmin(np.abs(stim_freq - float(f))) + 1 for f in annots.aux_note
-            ]
-            raw_events = np.zeros([1, n_samples])
-            #  annots.sample indicates the start of the trial
-            # of class "events_label"
-            for label, samploc in zip(events_label, annots.sample):
-                raw_events[0, samploc] = label
-            # append the data as another channel(stim) in the data
-            data = np.concatenate((data, raw_events), axis=0)
             if self.code == "SSVEP MAMEM3":
-                ch_names = record.sig_name
+                m = loadmat(fpath)
+                ch_names = [e[0] for e in m["info"][0, 0][9][0]]
```

Comment:
You have to love mat files and how "obvious" their acces is... :)

Reply:
Yes, but I'd rather deal with mat file than physionet format and WFDB that comes with plot and specific download function ^^